arangod/Graph/GraphManager.cpp

```diff
@@ -765,11 +767,98 @@ Result GraphManager::readGraphByQuery(velocypack::Builder& builder,
         << "cannot read graphs from _graphs collection";
   }
 
-  builder.add(VPackValue(VPackValueType::Object));
-  builder.add("graphs", graphsSlice);
+  builder.openObject();
+  if (!onlyHash) {
+    /*
+     * Will create {"graphs": [...]}
+     */
+    builder.add(VPackValue(StaticStrings::GraphsArray));
+    {
+      VPackArrayBuilder graphsArray(&builder);
+      for (VPackSlice graphSlice : VPackArrayIterator(graphsSlice)) {
+        builder.openObject();
+        auto res = writeGraphToBuilder(builder, graphSlice, details);
+        if (res.fail()) {
+          builder.close();
+          return res;
+        }
+        builder.close();
+      }
+    }
+  } else {
+    /*
+     * Will create {"checksum": "<md5-of-graphs>"}
+     */
+    writeGraphsChecksumsToBuilder(builder, graphsSlice);
+  }
   builder.close();
 
-  return Result(TRI_ERROR_NO_ERROR);
+  return {TRI_ERROR_NO_ERROR};
+}
+
+Result GraphManager::writeGraphsChecksumsToBuilder(
+    VPackBuilder& builder, VPackSlice const& graphsSlice) const {
+  std::set<std::string> checksums;
+  auto checksumBuilder = [](VPackBuilder& builder,
+                            std::string const& checksum) {
+    TRI_ASSERT(builder.isOpenObject());
+    if (checksum.length() > 0) {
+      builder.add(StaticStrings::GraphChecksum,
+                  VPackValue(arangodb::rest::SslInterface::sslMD5(checksum)));
+    } else {
+      builder.add(StaticStrings::GraphChecksum, VPackValue(checksum));
+    }
+  };
+
+  TRI_ASSERT(graphsSlice.isArray());
+  for (VPackSlice graphSlice : VPackArrayIterator(graphsSlice)) {
+    auto res = lookupGraphByName(
+        graphSlice.get(StaticStrings::GraphName).copyString());
+    if (res.fail()) {
+      return res.result();
+    }
+    auto& graph = res.get();
+    // collect each checksum per graph
+    graph->graphForClientOnlyHash(checksums, _vocbase);
+  }
+
+  if (checksums.empty()) {
+    checksumBuilder(builder, "");
+  } else {
+    // calculate "superHash" out of all graphs (if at least one has been found)
```

Comment:
This sounds complicated.
Can we maybe use a numeric hash value instead of a string based one?
We could sum up all hashes to get a single combined hash. 
In this case we could actually XOR the hashes as well. As we have guaranteed that no two graphs are actually identical, excluding the issue why XOR is worse than ADD on hash combinations